src/date.rs

```diff
@@ -81,6 +81,8 @@ pub struct Date {
 }
 
 impl Date {
+    /// Allocates a GDate and initializes it to a sane state. The new date will be cleared
```

Comment:
Just an observation: ideally these types would not exist and the foreign ones would be converted to/from one of the Rust date/time types.